libuavcan/src/transport/uc_can_acceptance_filter_configurator.cpp

```diff
@@ -0,0 +1,228 @@
+        UAVCAN_TRACE("CanAcceptanceFilterConfigurator::loadInputConfiguration()", "cfg.MK [%u] = %d", i,
+                     multiset_configs_.getByIndex(i)->mask);
+    }
+#endif
+
+    return 0;
+}
+
+int16_t CanAcceptanceFilterConfigurator::computeConfiguration()
+{
+    const uint16_t acceptance_filters_number = getNumFilters();
+    UAVCAN_ASSERT(multiset_configs_.getSize() != 0);
+
+    while (acceptance_filters_number < multiset_configs_.getSize())
+    {
+        uint16_t i_rank = 0, j_rank = 0;
+        uint8_t best_rank = 0;
+
+        const uint16_t multiset_array_size = static_cast<uint16_t>(multiset_configs_.getSize());
+
+        for (uint16_t i_ind = 0; i_ind < multiset_array_size - 1; i_ind++)
+        {
+            for (uint16_t j_ind = static_cast<uint8_t>(i_ind + 1); j_ind < multiset_array_size; j_ind++)
+            {
+                CanFilterConfig temp_config = mergeFilters(*multiset_configs_.getByIndex(i_ind),
+                                                          *multiset_configs_.getByIndex(j_ind));
+                uint8_t rank = countBits(temp_config.mask);
+                if (rank > best_rank)
+                {
+                    best_rank = rank;
+                    i_rank = i_ind;
+                    j_rank = j_ind;
+                }
+            }
+        }
+
+        *multiset_configs_.getByIndex(j_rank) = mergeFilters(*multiset_configs_.getByIndex(i_rank),
+                                                             *multiset_configs_.getByIndex(j_rank));
+        multiset_configs_.removeFirst(*multiset_configs_.getByIndex(i_rank));
+    }
+
+    UAVCAN_ASSERT(acceptance_filters_number >= multiset_configs_.getSize());
+
+    return 0;
+}
+
+int16_t CanAcceptanceFilterConfigurator::applyConfiguration(void)
+{
+    CanFilterConfig filter_conf_array[MaxCanAcceptanceFilters];
+    const unsigned int filter_array_size = multiset_configs_.getSize();
+
+    if (filter_array_size > MaxCanAcceptanceFilters)
+    {
+        UAVCAN_ASSERT(0);
+        return -ErrLogic;
+    }
+
+    for (uint16_t i = 0; i < filter_array_size; i++)
+    {
+        CanFilterConfig temp_filter_config = *multiset_configs_.getByIndex(i);
+
+        filter_conf_array[i] = temp_filter_config;
+    }
+
+    ICanDriver& can_driver = node_.getDispatcher().getCanIOManager().getCanDriver();
+    for (uint8_t i = 0; i < node_.getDispatcher().getCanIOManager().getNumIfaces(); i++)
+    {
+        ICanIface* iface = can_driver.getIface(i);
+        if (iface == NULL)
+        {
+            return -ErrDriver;
+        }
+        int16_t num = iface->configureFilters(reinterpret_cast<CanFilterConfig*>(&filter_conf_array),
+                                              static_cast<uint16_t>(filter_array_size));
+        if (num < 0)
+        {
+            return -ErrDriver;
+        }
+    }
+
+    return 0;
+}
+
+int CanAcceptanceFilterConfigurator::configureFilters()
+{
+    if (getNumFilters() == 0)
```

Comment:
Alternatively, you could store the number of filters there, and pass the value into `computeConfiguration()`.